### PR TITLE
Atomic sites

### PR DESCRIPTION
**Pull request by plt-amy**
Starting with properties of pushouts of sets so we can characterise the atomic topology on `FinInj ^op`.

### CODE REVIEW

**Comment by Lavenza** [comment]
[Pull request preview](https://preview.1lab.dev/448)
##### Changed pages

  * [Data.Set.Coequaliser](https://preview.1lab.dev/448/Data.Set.Coequaliser.html) (+33, -0)
  * [Data.Fin.Finite](https://preview.1lab.dev/448/Data.Fin.Finite.html) (+22, -0)

File: src/Data/Set/Pushout.lagda.md [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+-->
+
+We then have the two decoding functions. In the case with matched
+endpoints, we have exactly our original goal: injectivity of
+`incr`{.Agda}.
+
+```agda
+  decodeᵣ : injective incr
+  decodeᵣ {a} p = transport (λ i → ⌞ code a (p i) ⌟) (lift refl) .lower
+```
```
Comment (ncfavier) [inline src/Data/Set/Pushout.lagda.md:185]: This is *en*coding, right? You give it a path in the pushout and it gives you a code for that path.

File: src/Data/Set/Pushout.lagda.md [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+A$, constructed by swapping the constructors.
+
+```agda
+swap-pushout : Pushout f g → Pushout g f
+swap-pushout (incl x) = incr x
+swap-pushout (incr x) = incl x
+swap-pushout (glue x i) = glue x (~ i)
+swap-pushout (squash x y p q i j) =
+  let f = swap-pushout in squash (f x) (f y) (λ i → f (p i)) (λ i → f (q i)) i j
+```
```
Comment (ncfavier) [inline src/Data/Set/Pushout.lagda.md:51]: This this doesn't appear in the prose, maybe it should be hidden?

File: src/Cat/Instances/Sheaves/Omega.lagda.md [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+```
+-->
+
+```agda
+    in Ω-ua (λ h∈S → cT h (rem₂' h∈S)) (λ h∈T → cS h (rem₃ h∈T))
+```
+
+Now we have to show that a family $S$ of compatible closed sieves over a
+$J(U)$-covering sieve $R$ can be uniquely patched to a closed sieve on
+$U$. This is the sieve which is defiend to contain $g : V \to U$
```
Comment (ncfavier) [inline src/Cat/Instances/Sheaves/Omega.lagda.md:123]:
```suggestion
$U$. This is the sieve which is defined to contain $g : V \to U$
```

File: src/Cat/Diagram/Coproduct/Copower.lagda.md [hunk 4]
```diff
@@ -120,3 +123,46 @@ cocomplete→copowering colim = Copowers.Copowering λ S F →
   Colimit→IC _ (is-hlevel-suc 2 (S .is-tr)) F (colim _)
 ```
 -->
+
+## Constant objects
+
+If $\cC$ has a terminal object $\star$, then the copower $S \times
+\star$ is referred to as the **constant object** on $S$. By simplifying
+the universal power of the copower, we obtain that constant objects
```
Comment (ncfavier) [inline src/Cat/Diagram/Coproduct/Copower.lagda.md:131]:
```suggestion
the universal property of the copower, we obtain that constant objects
```

File: src/Cat/Diagram/Coproduct/Copower.lagda.md [hunk 5]
```diff
@@ -120,3 +123,46 @@ cocomplete→copowering colim = Copowers.Copowering λ S F →
...
+assemble into a functor left adjoint to $\cC(\star,-)$.^[This right
+adjoint is often called the **global sections functor**]
```
Comment (ncfavier) [inline src/Cat/Diagram/Coproduct/Copower.lagda.md:133]:
```suggestion
adjoint is often called the **global sections functor**.]
```

File: src/Cat/Diagram/Coproduct/Copower.lagda.md [hunk 6]
```diff
@@ -120,3 +123,46 @@ cocomplete→copowering colim = Copowers.Copowering λ S F →
   Colimit→IC _ (is-hlevel-suc 2 (S .is-tr)) F (colim _)
 ```
 -->
+
+## Constant objects
```
Comment (ncfavier) [inline src/Cat/Diagram/Coproduct/Copower.lagda.md:127]: Where is this terminology from? I'd expect those to be called "discrete objects" like in the cohesion picture.

Comment (plt-amy) [reply src/Cat/Diagram/Coproduct/Copower.lagda.md:127]: It seems pretty common to refer to the left adjoint of Γ as assigning "constant sheaves"; I just generalised that. fwiw here's someone using ["constant objects"](https://mathoverflow.net/questions/277722/characterize-constant-objects-in-the-internal-language-of-a-topos) in the wild.

File: src/Cat/Site/Grothendieck.lagda.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+```agda
+module Cat.Site.Grothendieck where
+```
+
+# Grothendieck sites {defines="grothendieck-site grothendieck-coverage"}
+
+A **grothendieck topology** $J$ on a category $\cC$ is function
+assigning to each $U : \cC$ a predicate $J(U)$ on the [[sieves]] on $U$,
+satisfying the [[closure properties of sites]]. Every topology generates
+a topology (where the space of sieves that cover $U$ is the total space
```
[the remark this answers is not in the document]
Comment (plt-amy) [reply src/Cat/Site/Grothendieck.lagda.md:23]: well, that is true

[the remark this answers is not in the document]
Comment (plt-amy) [reply src/Cat/Site/Grothendieck.lagda.md:23]:
```suggestion
satisfying the [[closure properties of sites]]. Every coverage generates
a topology (where the space of sieves that cover $U$ is the total space
```

File: src/Data/Set/Pushout.lagda.md (hunk 1 above)
Comment (plt-amy) [reply src/Data/Set/Pushout.lagda.md:185]: i guess that's true.

File: src/Cat/Site/Grothendieck.lagda.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (ncfavier) [reply src/Cat/Site/Grothendieck.lagda.md]: Wait isn't it the opposite? The rest of the sentence says every site gives a topology.

File: src/Cat/Diagram/Coproduct/Copower.lagda.md (hunk 6 above)
Comment (ncfavier) [reply src/Cat/Diagram/Coproduct/Copower.lagda.md:127]: Fair enough.

File: src/Cat/Site/Grothendieck.lagda.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (plt-amy) [reply src/Cat/Site/Grothendieck.lagda.md]: fuck i should've fixed this in my text editor

**Review by ncfavier — APPROVED** [review]

**Pull request merged**